nni/runtime/log.py

```diff
@@ -7,14 +7,149 @@
 import sys
 from datetime import datetime
 from io import TextIOBase
-from logging import FileHandler, Formatter, Handler, StreamHandler
+from logging import FileHandler, Formatter, Handler, LogRecord, Logger, StreamHandler
 from pathlib import Path
+import string
 from typing import Optional
 
 import colorama
 
 from .env_vars import dispatcher_env_vars, trial_env_vars
 
+_global_logging_enabled: bool = False
+
+_root_logger: Logger = logging.getLogger()
+logging.getLogger('nni').setLevel(logging.INFO)
+
+_handlers: dict[str, Handler] = {}
+
+colorama.init()
+_colorful: dict[str, str] = colorama.Fore.__dict__
+_colorless: dict[str, str] = {key: '' for key in _colorful.keys()}
+
+def start_stdout_logging() -> None:
+    """
+    Register the stdout handler.
+
+    This function should be invoked on importing nni.
+
+    It is safe to call it multiple times.
+    """
+    if '_stdout_' in _handlers:
+        return
+
+    handler = StreamHandler(sys.stdout)
+    handler.addFilter(_nni_filter)
+    handler.setFormatter(_StdoutFormatter())
+
+    _handlers['_stdout_'] = handler
+    _root_logger.addHandler(handler)
```

Comment:
I think adding a handler to `logging.getLogger('nni')` enables log-handling on every logger prefixed with NNI. There's no need to manipulate the root logger. Maybe I'm wrong.

Reply:
The filter was introduced by previous design. The main reason I kept it is `enable_global_logging()`. In current implementation it only needs to set a switch variable. Otherwise it will have considerably complex logic.
If you think it's bad to use root logger + custom filter, I can change it anyway.